### PR TITLE
Set up firebase

### PR DESCRIPTION
**Pull request by bt-s**
A short summary of what has been created:
- Firebase set-up
- Created SignUp, SignIn, SignOut, PasswordChange, PasswordForget and Account components
- Set up code for protection auth-only routes

I understand if this code is not immediately obvious. 
If you want to know how all of this works in depth, feel free to ask me or consult the following source:
- https://www.robinwieruch.de/complete-firebase-authentication-react-tutorial/

Our account at Firebase can be found following the link below:
- https://console.firebase.google.com/project/bookio/authentication/users

 I have created two test accounts. The easiest one for testing has username 'name@test.com' with password 'password'.

Don't forget to run $ npm install to download all dependencies and to add the Firebase API key in the /APIKeys folder.

EDIT:
Instead of using the context provider we should use Redux. I will look into this later this week, hopefully.

### CODE REVIEW

File: src/components/Navbar/Navbar.js [hunk 1]
```diff
@@ -8,13 +8,25 @@ import {
...
+import {AuthUserContext} from '../Session/Session';
 import Button from '../Button/Button';
+import SignOut from '../SignOut/SignOut';
+
+const NavbarAuth = () => <SignOut />;
 
-class NavBar extends React.Component {
+const NavbarNonAuth = () => (
+  <React.Fragment>
+    <Link to={ROUTES.LOG_IN}>Sign in</Link>;
```
Comment (Envl) [inline src/components/Navbar/Navbar.js:24]: As SignIn and SignUp are integrated into one component in the design, we only need one React Component for them as well. But we can change that later.

File: src/components/Navbar/Navbar.js [hunk 2]
```diff
@@ -30,43 +42,72 @@ class NavBar extends React.Component {
...
   }
 }
 
-NavBar.propTypes = {
+Navbar.propTypes = {
   logo: PropTypes.string
 };
 
-NavBar.defaultProps = {
+Navbar.defaultProps = {
```
Comment (Envl) [inline src/components/Navbar/Navbar.js:109]: We could use either React Hooks or @Decorators to improve this part, setting us propTypes and defaultProps at the end of the file, which is not convenient.

File: src/components/PasswordForget/PasswordForget.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+const INITIAL_STATE = {
+  email: '',
+  error: null
+};
+
+class PasswordForgetFormBase extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {...INITIAL_STATE};
```
Comment (Envl) [inline src/components/PasswordForget/PasswordForget.js:24]: I think `this.state = { ...INITIAL_STATE }` is the same thing as` this.state = INITIAL_STATE`

**Review by Envl — COMMENTED** [review]
Good work Bas!

File: src/components/PasswordForget/PasswordForget.js (hunk 3 above)
Comment (bt-s) [reply src/components/PasswordForget/PasswordForget.js:24]: I am not too sure, but will have a look.

File: src/components/Navbar/Navbar.js (hunk 2 above)
Comment (bt-s) [reply src/components/Navbar/Navbar.js:109]: I will look into this right now.

File: src/components/Navbar/Navbar.js (hunk 1 above)
Comment (bt-s) [reply src/components/Navbar/Navbar.js:24]: Sure, we can change this later!

**Review by Envl — APPROVED** [review]

**Pull request merged**